elements/dash-accordion/dash-accordion.js

```diff
@@ -0,0 +1,529 @@
+ * can be toggled between expanded and collapsed state.
+ *
+ * The panels have the classes `expanded` or `collapsed` assigned to them
+ * depending on their state. However, these classes are only applied when the
+ * the animation is done. For most purposes, the `aria-expanded` attribute of
+ * the associated heading should be used as an indicator of the current state.
+ *
+ * If JavaScript is disabled, all panels are shown interleaved with the
+ * respective headings.
+ */
+(function() {
+  /**
+   * Define key codes to help with handling keyboard events.
+   */
+  const KEYCODE = {
+    DOWN: 40,
+    LEFT: 37,
+    RIGHT: 39,
+    UP: 38,
+    HOME: 36,
+    END: 35,
+    SPACE: 32,
+    ENTER: 13,
+  };
+
+  /**
+   * `DashAccordion` is a container element for headings and panels.
+   *
+   * All children of `<dash-accordion>` should be either
+   * `<dash-accordion-heading>` or `<dash-accordion-panel>`. This element is
+   * stateless, meaning that no values are cached and therefore, changes during
+   * runtime are incorporated.
+   */
+  class DashAccordion extends HTMLElement {
+    constructor() {
+      super();
+    }
+
+    connectedCallback() {
+      // The headings emit elements depending on how they get interacted with.
+      this.addEventListener('accordion-expand', this._onAccordionExpand);
+      this.addEventListener('accordion-collapse', this._onAccordionCollapse);
+      this.addEventListener('accordion-previous', this._onAccordionPrevious);
+      this.addEventListener('accordion-next', this._onAccordionNext);
+      this.addEventListener('accordion-first', this._onAccordionFirst);
+      this.addEventListener('accordion-last', this._onAccordionLast);
+
+      // Before the elements starts booting, it waits for
+      // the both `<dash-accordion-heading>` and `<dash-accordion-panel>` to
+      // load.
+      Promise.all([
+        customElements.whenDefined('dash-accordion-heading'),
+        customElements.whenDefined('dash-accordion-panel'),
+      ]).then(_ => {
+        // Acquire all headings and panels inside the element
+        const headings = this._allHeadings();
+        const panels = this._allPanels();
+        // If there are no headings, there is no way to activate panels.
+        // Abort.
+        if (headings.length === 0) return;
+
+        // Set up `aria-controls` and `aria-labelledby` on headings and panels.
+        headings.forEach(heading => {
+          const panel = heading.nextElementSibling;
+          if(!panel || panel.tagName !== 'DASH-ACCORDION-PANEL') {
+            console.error(`Tab #${heading.id} is not sibling` +
+              `of a <dash-accordion-panel>`);
+            return;
+          }
+
+          heading.setAttribute('aria-controls', panel.id);
+          panel.setAttribute('aria-labelledby', heading.id);
+        });
+
+        // Set all the panels to the collapsed state. This is done directly
+        // (and not using `_collapsePanel()`) to skip the animation.
```

Comment:
Yeah, that’s better. I might opt for collapsePanel({animate: false}) as it is more descriptive